lib/td/client/api.rb

```diff
@@ -300,17 +301,17 @@ def do_get(url, params=nil, &block)
         else
           response = client.get(target, params, header)
 
-          validate_content_length!(response, response.body.bytesize) if @ssl
-        end
+          status = response.code
+          # retry if the HTTP error code is 500 or higher and we did not run out of retrying attempts
+          if !block_given? && status >= 500 && cumul_retry_delay < @max_cumul_retry_delay
```

Comment:
This is in `else` for `if block`, so `!block_given?` is always true. We don't need to check it.